### DISCUSSION TITLE
What does that means "normal mode" is there others modes ?

### DISCUSSION THREAD

**Discussion by DzzD** [supermerill/SuperSlicer#1885] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/12198339/141761762-75e2c10f-67bb-40c7-bdd4-213aead6443e.png)

**Comment by neophyl** [comment, marked as answer]
Some printer firmware supports stealth mode (prusa mk3 to name the most common one).  Under Printer Settings>General theres a tickbox for supports stealth.  Also under Machine limits you will notice a second set of entries for the speeds/accelerations  where you can set stealth values.
Stealth mode uses more conservative speed/acceleration settings to in effect quiet down the printer somewhat.  As such the print time usually takes longer, so in the gcode output for the time remaining there will be 2 sets, one for normal printing and a second for if the printer is in stealth.  
So with a mk3 you can set the printer to normal mode and print but then if you want it to be less noisy (say if you are going to bed) you can then just select stealth and the printer will use the slower speeds that were passed to it in the gcode.

**Reply by DzzD** [reply]
clear, thanks !

**Comment by DzzD** [comment]
And that's another good reason to implements options/variant selector feature https://github.com/supermerill/SuperSlicer/issues/1874, we could create for a printer different modes with different machine limits without relying on board type.

**Comment by neophyl** [comment]
Not really no.  You have to have PRINTER FIRMWARE support to allow switching between 2 sets of values.  You can set and pass them to any printer but if it can't use them then there no point, although it doesn't hurt to pass them usually.

You could set up all the profiles you want with different speeds and then run the "Quiet greeblie" file or the "normal greeblie" file.  That doesn't really help you switch in the middle of a print though does it.

Basically Prusa implemented it for their firmware and added all the specialised custom bits in PS to support it which as SuSi is based on it means they are available too.  I guess if other manufacturers or baseline Marlin want to start using it too then great but until then its a niche use for those that use Prusa printers.

**Reply by DzzD** [reply]
Not sure I understand all what you said. 

What I suggested in the issue is to create differents mode/option to overide one or more parameters in a profile (printer, filament or print) (a little like simplify3d do with high/medium/fast but for all parameters)

**Reply by neophyl** [reply]
Creating different options is fine but they are all applied at the time of slicing.  SO your gcode contains one set of speeds/extrusions etc.  Once you have sliced and started a print you cant change the operation as thats all the printer is designed to do.  It blindly and dumbly follows the gcode, nothing else.  Doesnt matter what your slicer is designed to do.

What the current stealth mode does is change the PRINTER FIRMWARE to support 2 sets of speeds and then the slicer has been modified to ADD that second set into the gcode if the option is enabled.  That way the PRINTER can change/choose between the 2 sets on the fly, in the middle of a print.
You dont have to 'apply' stealth options, once set up they are always there.

Unless a printer is designed to support it then theres not a lot you can do.

I hope that makes it clearer.

**Reply by DzzD** [reply]
Yes, thanks, I did understand the fact that it was at runtime. But having those option at compile time would be great too, that was my point in another thread.
